CoinChange.java

```diff
@@ -0,0 +1,64 @@
+// Time Complexity : O(A * C) where A = Amount and C = no of coins
+// Space Complexity : O(A) where A = Amount 
+// Did this code successfully run on Leetcode : Yes
+// Any problem you faced while coding this :
+
+
+// Your code here along with comments explaining your approach
+//1. We take a 2D matrix dp and cache the values of the subproblems.
```

Comment:
Please use a brief overview in English sentences and not a detailed implementation in pseudocode.